code/modules/surgery/bodyparts/dismemberment.dm

```diff
@@ -276,15 +276,24 @@
 		return FALSE
 	return TRUE
 
-///Attach src to target mob if able, returns FALSE if it fails to.
-/obj/item/bodypart/proc/try_attach_limb(mob/living/carbon/new_limb_owner, special)
+/*
+ * Attach src to target mob if able, returns FALSE if it fails to.
+ * Arguments:
+ * special - The limb is being hotswapped or removed without side effects for some reason
+ * lazy - The owner is currently initializing, so we don't need to call any update procs
+ */
+/obj/item/bodypart/proc/try_attach_limb(mob/living/carbon/new_limb_owner, special, lazy)
 	if(!can_attach_limb(new_limb_owner, special))
 		return FALSE
 
-	SEND_SIGNAL(new_limb_owner, COMSIG_CARBON_ATTACH_LIMB, src, special)
-	SEND_SIGNAL(src, COMSIG_BODYPART_ATTACHED, new_limb_owner, special)
+	SEND_SIGNAL(new_limb_owner, COMSIG_CARBON_ATTACH_LIMB, src, special, lazy)
+	SEND_SIGNAL(src, COMSIG_BODYPART_ATTACHED, new_limb_owner, special, lazy)
 	new_limb_owner.add_bodypart(src)
 
+	if(lazy)
+		SEND_SIGNAL(new_limb_owner, COMSIG_CARBON_POST_ATTACH_LIMB, src, special, lazy)
```

Comment:
I'd make more sense to be sorting here wouldn't it? 
Or just make lazy skip the stuff you want but continue down to still sort & send signal instead of sending it in 2 different places

Reply:
We'd be sorting all the limbs after every limb being added, which is not necessary.

Reply:
My second point still stands, from what it looks like this will prevent persistent scars from loading?

Reply:
Good point, will do